crates/driver/src/simulator/tenderly/mod.rs

```diff
@@ -19,26 +21,65 @@ pub struct Config {
 }
 
 impl Tenderly {
-    pub fn new(_config: Config) -> Self {
-        todo!()
+    pub fn new(config: Config) -> Self {
+        let mut headers = reqwest::header::HeaderMap::new();
+        headers.insert(
+            reqwest::header::CONTENT_TYPE,
+            "application/json".parse().unwrap(),
+        );
+        headers.insert(reqwest::header::ACCEPT, "application/json".parse().unwrap());
+        Self {
+            client: reqwest::ClientBuilder::new()
+                .default_headers(headers)
+                .build()
+                .unwrap(),
+            config,
+        }
     }
 
     pub async fn simulate(
         &self,
-        _tx: &eth::Tx,
-        _access_list: &eth::AccessList,
-        _speed: Speed,
-    ) -> Result<eth::Simulation, Error> {
-        todo!()
+        tx: &eth::Tx,
+        access_list: &eth::AccessList,
+        generate_access_list: GenerateAccessList,
+    ) -> Result<super::Simulation, Error> {
+        let url = self.config.url.join("simulate").unwrap();
+        let res: dto::Response = self
+            .client
+            .post(url)
+            .json(&dto::Request {
+                network_id: self.config.network.to_string(),
```

Comment:
This should be a "network ID" or "network version" (result of `net_version` RPC call) and not a descriptive network name ("testnet"/"mainnet").

Personally, I would be for using `eth::ChainId` everywhere, and implementing a `eth::ChainId::network_id() -> &'static str` method (since a network ID/version is fixed for a chain ID).